### ISSUE TITLE
How to build v8 to work with this project?

### ISSUE THREAD

**Issue by Zeblote** [ncsoft/Unreal.js#178]
How are you creating the v8 binaries? I tried pretty much every combination of common settings in the args.gn file:

![image](https://user-images.githubusercontent.com/6764788/29696570-1ce1ad74-894a-11e7-9d96-c0aaf2e0ed7f.png)

But there are 2 problems:
- you have 4 v8_base_x.lib files, I only get a single v8_base.lib
- attempting to replace your libs with the new ones results in:
```v8_base.lib(api.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in SharedPCH.UnrealEd.h.obj```

**Comment by crocuis** [comment]
Have you read this? Https://github.com/ncsoft/Unreal.js/wiki/V8 :)

**Comment by Zeblote** [comment]
Oh! That's for building with gyp though, v8 wants to build with this new gn thing now... I'll see if it has similar options

**Comment by Zeblote** [comment]
I tried using similar gn settings, but it's not successful so far. The setting v8_enable_inspector doesn't actually exist in gyp or gn.

```
is_debug = false
target_cpu = "x64"
v8_use_snapshot = false
v8_static_library = true
is_component_build = true
v8_enable_i18n_support = false
treat_warnings_as_errors = false
```

The lib is huge now! v8_base.lib is over 800MB... can't be right.
Still causes hundreds of errors also:
![image](https://user-images.githubusercontent.com/6764788/29715404-906fe6d6-89a7-11e7-9882-918ac459ef7c.png)

**Comment by Zeblote** [comment]
Next try:
```
is_debug = false
target_cpu = "x64"
v8_use_snapshot = false
v8_static_library = true
is_component_build = false
v8_enable_i18n_support = false
treat_warnings_as_errors = false
```
I also to v8/build/config/win/BUILD.gn and changed `configs = [ ":static_crt" ]` to `configs = [ ":dynamic_crt" ]` since the force_dynamic_crt setting doesn't exist anymore in gn.

The v8_base.lib is back to normal size now... 350MB
It also generated significantly less errors:
![image](https://user-images.githubusercontent.com/6764788/29715724-c6106e2c-89a8-11e7-9dd6-a1f4ae69c52b.png)
... but it still isn't working.

**Comment by crocuis** [comment]
Is this the same problem? Https://bugs.chromium.org/p/v8/issues/detail?id=6461

**Comment by Zeblote** [comment]
I don't think it is, building v8 works fine. The problem only appears when trying to link it in your plugin.

**Comment by crocuis** [comment]
I did not have a problem with gyp build. after I build with gn, I'll try to figure out what the problem is.

**Comment by crocuis** [comment]
I think the version of v8 you built seems to be higher than 6.0. Version 6.0 or later requires additional v8_builtins_setup.lib and v8_builtins_generators.lib.

**Comment by Zeblote** [comment]
Oh. I've added these 2 libs, but it is still not linking properly.
Now the number of errors is down to 2 though:
```
2>Module.V8.cpp.obj : error LNK2019: unresolved external symbol "public: static class std::unique_ptr<class v8_inspector::V8Inspector,struct std::default_delete<class v8_inspector::V8Inspector> > __cdecl v8_inspector::V8Inspector::create(class v8::Isolate *,class v8_inspector::V8InspectorClient *)" (?create@V8Inspector@v8_inspector@@SA?AV?$unique_ptr@VV8Inspector@v8_inspector@@U?$default_delete@VV8Inspector@v8_inspector@@@std@@@std@@PEAVIsolate@v8@@PEAVV8InspectorClient@2@@Z) referenced in function "public: __cdecl FInspector::FInspector(class v8::Platform *,int,class v8::Local<class v8::Context>)" (??0FInspector@@QEAA@PEAVPlatform@v8@@HV?$Local@VContext@v8@@@2@@Z)
2>v8_libbase.lib(stack_trace_win.obj) : error LNK2019: unresolved external symbol __imp_PathRemoveFileSpecW referenced in function "bool __cdecl v8::base::debug::`anonymous namespace'::InitializeSymbols(void)" (?InitializeSymbols@?A0x76313174@debug@base@v8@@YA_NXZ)
```
How does your args.gn file look? Or did you change something else to make it work?

**Comment by crocuis** [comment]
No. I found out that I was experiencing the same thing as your error when building v8 version 6.1 with GYP instead of GN. Assuming your error message, I think v8_enable_inspector is missing.

**Comment by Zeblote** [comment]
I searched the entire v8 folder for enable_inspector and the setting doesn't exist. Maybe it has been renamed.

**Comment by crocuis** [comment]
There was no problem in building with GYP. If it is a GN build only problem, Config is guessed to be a different problem

**Comment by Zeblote** [comment]
Ok, it seems to work if I also copy over `v8/out.gn/x64.release/obj/src/inspector/inspector.lib` and add `PublicAdditionalLibraries.Add("shlwapi.lib");` to `V8.Build.cs`.

Is there any advantage to building with GN then? I only wanted to use that because the v8 wiki says GYP is supposed to be deprecated and removed soon.

**Comment by crocuis** [comment]
When I built V8, I did not know about building V8 with GN. If you are good at building V8 with GN, please update to GN build at https://github.com/ncsoft/Unreal.js/wiki/V8.

**Comment by tank90fc** [comment]
I build v8(Version 6.2) with gn. UnrealEngine crashes when loading "UnrealJS plugin"
![image](https://user-images.githubusercontent.com/17718211/29903392-443fb3e8-8e36-11e7-91e6-05344daca909.png)
![image](https://user-images.githubusercontent.com/17718211/29903398-4c316efc-8e36-11e7-9d23-39a35854ab02.png)

**Comment by crocuis** [comment]
Since version v8 (Version 6.2) is not officially released, there seems to be a variety of internal problems. If the V8 API has changed or if it is necessary to implement it, I will add it later.

**Comment by hamkoo** [comment]
> I build v8(Version 6.2) with gn. UnrealEngine crashes when loading "UnrealJS plugin"
> ![image](https://user-images.githubusercontent.com/17718211/29903392-443fb3e8-8e36-11e7-91e6-05344daca909.png)
> ![image](https://user-images.githubusercontent.com/17718211/29903398-4c316efc-8e36-11e7-9d23-39a35854ab02.png)

I have the same error. Now, have you resolved it? @
